docs/basics/primitives.md

```diff
@@ -5,30 +5,195 @@ to describe the set of types that are native to its platform, we leverage and ov
 that Cassandra can natively understand.
 
 The `Primitive` trait has a simplistic implementation that contains all the necessary information to
-basically serialize and de-serialize a given type as it is retrieved from the database, and that's the 
+basically serialize and de-serialize a given type as it is retrieved from the database, and that's the
 basic primary goal. Whether it's a normal column or a user defined type or anything similar, this is
 the construct that de-serializes everything into familiar Scala types.
 
 
+### Default primitives and columns
+
+This is the list of available columns and how they map to C* data types. The columns
+here are nothing more than simple helpers and they do not actually contain any
+specialized implementation. That means it is the primitives doing all the work,
+columns are just a nice way for you to write DSL code that's readable.
+
+| phantom columns               | Java/Scala type           | Cassandra type    |
+| ---------------               |-------------------        | ----------------- |
+| BlobColumn                    | java.nio.ByteBuffer       | blog              |
+| BigDecimalColumn              | scala.math.BigDecimal     | decimal           |
+| BigIntColumn                  | scala.math.BigInt         | varint            |
+| BooleanColumn                 | scala.Boolean             | boolean           |
+| DateColumn                    | java.util.Date            | timestamp         |
+| DateTimeColumn                | org.joda.time.DateTime    | timestamp         |
+| DoubleColumn                  | scala.Double              | double            |
+| EnumColumn                    | scala.Enumeration         | text              |
+| FloatColumn                   | scala.Float               | float             |
+| IntColumn                     | scala.Int                 | int               |
+| InetAddressColumn             | java.net.InetAddress      | inet              |
+| LongColumn                    | scala.Long                | long              |
+| StringColumn                  | java.lang.String          | text              |
+| UUIDColumn                    | java.util.UUID            | uuid              |
+| TimeUUIDColumn                | java.util.UUID            | timeuuid          |
+| ListColumn[Type]              | immutable.List[Type]      | list<type>        |
+| SetColumn[Type]               | immutable.Set[Type]       | set<type>         |
+| MapColumn[Type, Type]         | immutable.Map[Type, Type] | map<type, type>   |
+| CounterColumn                 | scala.Long                | counter           |
+| StaticColumn&lt;type&gt;      | &lt;type&gt;              | type static       |
+
+
+<a id="optional-primitive-columns">Optional primitive columns</a>
+===================================================================
+
+Optional columns allow you to set a column to a ```null``` or a ```None```.
```

Comment:
Optional columns allow you to set a column to a ```null``` or a ```None```.
The outcome is that instead of a ```T``` you get an ```Option[T]``` and you can ```match, fold, flatMap, map```.
The ```Optional``` part is handled at a DSL level.